torch/testing/_internal/common_methods_invocations.py

```diff
@@ -602,6 +602,31 @@ def sample_inputs_addmm(op_info, device, dtype, requires_grad, **kwargs):
     else:
         return (input, )
 
+def sample_inputs_addcmul(op_info, device, dtype, requires_grad, **kwargs):
+    test_cases = [((S, S), (S, S), (S, S)),
```

Comment:
```
test_cases = (
  ((S, S), (S, S), (S, S)),
  ((S, S), (S, 1), (1, S)),
  ((1,), (S, S, 1), (1, S)),
  ((), (), ()),
  ((S, S), (), ()),
  ((), (S, S, 1), (1, S)),
)
```

Your approach is correct, but because you're (cleverly) applying the value kwarg programmatically below this doesn't need the repeated shape cases like the method_tests() entries did.

Reply:
Ho, I'm really bad at de-duplication... haha